[Bug]: fix: RegexEntityConfig loads with system locale encoding instead of UTF-8, breaking tests on Windows

### Bug Description

In regex_entity_config.py, the configuration file regex_entity_config.json is opened using open(self.config_path, "r") without an explicit encoding parameter.

On Windows, Python defaults to using the system's locale encoding (usually CP1252) instead of UTF-8. Since the configuration file defines Unicode patterns (e.g. the Euro sign € in the MONEY regex), loading it on Windows corrupts these characters in memory. This leads to test_extract_money failing in regex_entity_extraction_test.py.

### Steps to Reproduce

1. Set up the development environment on a Windows machine.
2. Run the command: uv run pytest cognee/tests/unit/entity_extraction/regex_entity_extraction_test.py -v
3. Observe the failure on test_extract_money.

### Expected Behavior

The regex configurations load successfully without encoding errors and the test suite passes on all operating systems (including Windows).

### Actual Behavior

The regex pattern for MONEY fails to extract values using Unicode currency symbols (like €) on Windows. This happens because the configuration JSON is loaded using system default encoding (CP1252), which corrupts the regex pattern in memory and causes test_extract_money to fail.

### Environment

- **OS**: Windows 11
- **Python version**: 3.14.5
- **Cognee version**: 1.2.1-local
- **LLM Provider**: None (Unit Test)
- **Database**: SQLite / LanceDB (default configuration))

### Logs/Error Messages

```shell
_____________________________ test_extract_money ______________________________

regex_extractor = <cognee.tasks.entity_completion.entity_extractors.regex_entity_extractor.RegexEntityExtractor object at 0x000002351BF38050>

    @pytest.mark.asyncio
    async def test_extract_money(regex_extractor):
        """Test extraction of monetary amounts."""
        text = (
            "The product costs $1,299.99 or €1.045,00 depending on your region. "
            "Malformed values like $1,234.567,89 and €1.234,567.89 should be ignored. "
            "Item 123 is in room 45."
        )
        entities = await regex_extractor.extract_entities(text)
    
        money_entities = [e for e in entities if e.is_a.name == "MONEY"]
        money_values = [e.name for e in money_entities]
    
>       assert len(money_entities) == 2
E       AssertionError: assert 1 == 2
E        +  where 1 = len([Entity(id=UUID('64843420-c982-5310-88ad-bd1d8e1852bd'), description='Monetary amount: $1,299.99')])

cognee\tests\unit\entity_extraction\regex_entity_extraction_test.py:96: AssertionError
```

### Additional Context

The issue is resolved by explicitly specifying encoding="utf-8" in open() when loading the config JSON inside cognee/tasks/entity_completion/entity_extractors/regex_entity_config.py.

I have the working fix and would like to submit a PR for it. Please assign it to me!

### Pre-submission Checklist

- [x] I have searched existing issues to ensure this bug hasn't been reported already
- [x] I have provided a clear and detailed description of the bug
- [x] I have included steps to reproduce the issue
- [x] I have included my environment details

--- a/cognee/tasks/entity_completion/entity_extractors/regex_entity_config.py
+++ b/cognee/tasks/entity_completion/entity_extractors/regex_entity_config.py
@@ -40,7 +40,7 @@ class RegexEntityConfig:
     def _load_config(self) -> None:
         """Load and process the configuration from the JSON file."""
         try:
-            with open(self.config_path, "r") as f:
+            with open(self.config_path, "r", encoding="utf-8") as f:
                 config_list = json.load(f)
         except FileNotFoundError:
             logger.error(f"Config file not found: {self.config_path}")

